x-pack/plugins/security_solution/server/lib/detection_engine/migrations/create_signals_migration_index.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ElasticsearchClient } from 'src/core/server';
+
+/**
+ * Creates the destination index to be used during the migration of a
+ * given signals index.
+ *
+ * The destination index's name is determined by adding a suffix of
+ * `-r${templateVersion}` to the source index name
+ *
+ * @param esClient An {@link ElasticsearchClient}
+ * @param index name of the source signals index
+ * @param version version of the current signals template/mappings
+ *
+ * @returns the name of the created index
+ */
+export const createSignalsMigrationIndex = async ({
+  esClient,
+  index,
+  version,
+}: {
+  esClient: ElasticsearchClient;
+  index: string;
+  version: number;
+}): Promise<string> => {
+  const paddedVersion = `${version}`.padStart(6, '0');
+  const destinationIndexName = `${index}-r${paddedVersion}`;
```

Comment:
Here's how we're naming our destination indexes, so e.g. migrating `.siem-signals-default-000002` to template 14 would reindex into `.siem-signals-default-000002-r000014`.

Note that this is additive as well, so if on a subequent upgrade the user migrated this index _again_, it'd be something like `.siem-signals-default-000002-r000014-r000021` etc. This allows us to track all the migrations that have been performed on a given index, but it also places a limit on the number of times they can migrate before hitting the index name limit.

Reply:
Do we need the `0` padding on the version numbers? Could reduce some noise and allow them to migrate more before hitting the name length limit.

Reply:
The padding was added to assist sorting these indexes by name. It's not strictly necessary and not being relied upon anywhere, so we could definitely remove it.